cmd/cluster-node-tuning-operator/main.go

```diff
@@ -321,6 +334,45 @@ func migratePinnedSingleNodeInfraStatus(cfg *rest.Config, scheme *apiruntime.Sch
 	return nil
 }
 
+func setupFeatureGates(ctx context.Context, config *rest.Config, operatorNamespace string) (featuregates.FeatureGate, error) {
+	// TODO is this version correct?
+	desiredVersion := version.Version
+	missingVersion := "0.0.1-snapshot"
+
+	configClient, err := configclient.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	k8sClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	controllerRef, err := events.GetControllerReferenceForCurrentPod(ctx, k8sClient, operatorNamespace, nil)
+	if err != nil {
+		return nil, err
+	}
+	eventRecorder := events.NewKubeRecorder(k8sClient.CoreV1().Events(metav1.NamespaceNone), "performance-profile-controller", controllerRef)
```

Comment:
can we get this from controller-runtime? (line 188 above)

Reply:
I tried that, but the interfaces are completely different